src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb

```diff
@@ -405,12 +406,23 @@ Namespace Microsoft.VisualStudio.Editors.DesignerFramework
         Private Function TryGetCustomToolPropertyValue(ByRef value As String) As Boolean
             value = Nothing
 
-            Dim customToolProperty As EnvDTE.Property = DTEUtils.GetProjectItemProperty(_projectItem, DTEUtils.PROJECTPROPERTY_CUSTOMTOOL)
-            If customToolProperty IsNot Nothing Then
-                Dim customToolValue As String = TryCast(customToolProperty.Value, String)
-                value = customToolValue
-                Return True
-            End If
+            Try
+                Dim customToolProperty As EnvDTE.Property = DTEUtils.GetProjectItemProperty(_resxFileProjectItem, DTEUtils.PROJECTPROPERTY_CUSTOMTOOL)
```

Comment:
Can a custom tool be something else than a resx file? Wondering if we are aware of other scenarios this code might affect. Looking at the `DTEUtils.PROJECTPROPERTY_CUSTOMTOOL` makes me think we can have other items.

Reply:
In this try/catch we will only handle when the item id is invalid caused by some operation at the file level. This property should not change.